generate.py

```diff
@@ -36,50 +39,64 @@ def init_verbose_logger():
     logger.addHandler(ch)
 
 
-def print_about_ydk_page(ydk_root, py_api_doc_gen):
+def print_about_page(ydk_root, py_api_doc_gen):
     repo = Repo(ydk_root)
     remote = repo.remote().name
     branch = repo.active_branch.name
     url = repo.remote().url.split('://')[-1].split('.git')[0]
     commit_id = repo.rev_parse(remote + '/' + branch).hexsha
 
-    contents = ''
-
-    with open(os.path.join(py_api_doc_gen, 'about_ydk.rst'), 'r+w') as about_file:
-        contents = about_file.read()
-        contents = contents.replace('git clone repo-url', 'git clone https://{0}.git'.format(url))
-        contents = contents.replace('git checkout commit-id', 'git checkout {0}'.format(commit_id))
-
-    with open(os.path.join(py_api_doc_gen, 'about_ydk.rst'), 'w') as about_file:
-        about_file.write(contents)
-
-
-def generate_documentation(output_directory, ydk_root):
-
-    py_api_doc_gen = output_directory + '/python/docsgen'
-    py_api_doc = output_directory + '/python/docs_expanded'
-    execfile(os.path.join(output_directory, 'python', 'ydk', '_version.py'))
-
+    # modify about_ydk.rst page
+    for line in fileinput.input(os.path.join(py_api_doc_gen, 'about_ydk.rst'), 'r+w'):
+       if 'git clone repo-url' in line:
+           print line.replace('repo-url', 'https://{0}.git'.format(url)),
+       elif 'git checkout commit-id' in line:
+           print line.replace('commit-id', '{}'.format(commit_id))
+       else:
+           print line,
+
+
+def print_bundle_landing_page(py_api_doc_gen, bundle):
+    if bundle:
+        bundle_name = os.path.basename(bundle).rstrip('.json')
+        for line in fileinput.input(os.path.join(py_api_doc_gen, 'index.rst'), 'r+w'):
+            if '<bundle_name>' in line:
+                print line.replace('<bundle_name>', bundle_name),
+                print len(line) * '='
+            else:
+                print line,
+
+
+def get_release_version(output_directory):
+    version = ''
+    release = ''
+    setup_file = os.path.join(output_directory, 'setup.py')
+    with open(setup_file, 'r') as f:
+        for line in f:
+            if 'version=' in line or 'version =' in line:
+                rv = line[line.find('=') + 1:line.rfind(",")]
+                release = "release=" + rv
+                version = "version=" + rv[:rv.rfind(".")] + "'"
+    return release, version
```

Comment:
Sorry if I missed this, but where are we generating a version file (`__version__.py`) to store the version? Why are we reading setup.py to get the version/revision?